device/cluster.cpp

```diff
@@ -1,7 +1,7 @@
 // SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
 //
 // SPDX-License-Identifier: Apache-2.0
-#include "cluster.h"
+#include "umd/device/cluster.h"
```

Comment:
Why is `umd/device` not part of the build include path?

Reply:
Because we haven't discussed what internal access should look like.  This PR was demonstrating external access and putting all the public headers in a segregated place to distinguish external vs internal.  That aspect is more straightforward / objective.

Internal access is more subjective.

On the one hand, accessing the same way as an external consumer makes it obvious when we're using a public API -- does that matter?  For some projects maybe that's useful information and for others it's inconsequential.
It also means that moving a file from internal to external doesn't require changing the includes.  In a project highly in flux, perhaps that's a useful convenience.  But if structing include ordering to indicate proximity, then it'll need to be updated anyway.  And in a more well established project hoisting a module externally is very very rare in the grand scheme of things.

On the other hand, accessing like any other internal header separates on a different concern: I'm conveying that this is a header within this project, but I'm not affected nor do I care whether said header is ALSO available externally or not.  Not my concern.

I believe @blozano-tt prefers the former.  I prefer the latter.  But ultimately neither of us own UMD, so it's you and @broskoTT who should make a call re: what you prefer internal for your project.  I can help guide on the CMake change for whichever you prefer.